account_commission/models/account_move.py

```diff
@@ -114,6 +114,11 @@ def fields_view_get(
                 res["arch"] = etree.tostring(invoice_xml)
         return res
 
+    def unlink(self):
+        for invoice in self:
+            invoice.invoice_line_ids.settlement_id.state = "settled"
```

Comment:
Put docstring here about why is the overriding, and you can do it in one step:

```suggestion
        self.invoice_line_ids.settlement_id.write({"state": "settled"})
```

Reply:
done :)